### ISSUE TITLE
Add filter for an participating organisation that is specifically funding/implementing etc.

### ISSUE THREAD

**Issue by andylolz** [codeforIATI/iati-datastore#34]

**Comment by markbrough** [comment]
e.g. for participating org role `1` (Funding), this could look like:

Any preferences on whether this should be `participating-org.1` or `participating-org-1`?

participating-org.1="GB-GOV-1"
participating-org.1.text="Foreign, Commonwealth and Development Office"
participating-org.1.ref="GB-GOV-1"
participating-org.1.type="10"

**Comment by odscjames** [comment]
I'm going to suggest:

    participating-org-role-1
    participating-org-role-1.ref
    participating-org-role-1.text

Because to me just having the number is confusing - 1 what? Being used to JSON paths my first thought was "the 1st partciptaing org in the list?"

Also, we don't need .type right, as type is the role filter and we are already filtering by role?

**Comment by markbrough** [comment]
We do need `type` - this is a different attribute. We're filtering on the `role` attribute (funding/implementing/[etc](https://codelists.codeforiati.org/OrganisationRole/)), but we still want to be able to filter on the `type` attribute (government/NGO/[etc](https://codelists.codeforiati.org/OrganisationType/)):

https://github.com/codeforIATI/iati-datastore/blob/83ecdb12dbbe361a14ff335bf04ad13d025b92db/iati_datastore/iatilib/model.py#L189-L195

**Comment by odscjames** [comment]
> We do need type - this is a different attribute

Oh yes, I had got confused with a different role filter I think

**Comment by odscjames** [comment]
I'm not sure I totally understand the requirement here - can you check https://github.com/codeforIATI/iati-datastore/pull/342 and make sure I've got it right? If so I'll then add some tests before sending for review

**Comment by markbrough** [comment]
Looks good to me!

**Comment by odscjames** [comment]
I have added one test to the PR but this has raised an issue - I've seen data in the role column be "1" or "Funding".

Probably most real data is "1" - here's a partial data load on my laptop:
```

app=> select role, count(role) from participation group by role;
     role     | count 
--------------+-------
 3            | 11208
 2            |  9997
 Funding      |    65
 Implementing |    76
 Extending    |    58
 1            | 13178
 Accountable  |     2
 4            | 12227


```
But when you write tests with "role=cl.OrganisationRole.funding"  like the existing tests are, you get data with "Funding".

We'll need to clear this up before continuing ...

**Comment by markbrough** [comment]
Yeah so the text values (`Funding`, etc) are from v1.x data. In the database, when importing these values, we should probably map these to v2 values (`1`, etc)

**Comment by markbrough** [comment]
See here:
v1: https://codelists.codeforiati.org/api/1/json/en/OrganisationRole.json
v2: https://codelists.codeforiati.org/api/2/json/en/OrganisationRole.json

NB the `name` in the v2 data is the `code` in the v1 data.

**Comment by odscjames** [comment]
>  In the database, when importing these values, we should probably map these to v2 values (1, etc)

This seems sensible to me, and links up to an earlier conversation we had about how strict to be with data correctness (that one was about white spaces around Id's I think?).

I can do this as a separate PR first?

On the dev server with full data BTW 

```
iatidatastoreclassic=> select role, count(role) from participation group by role;
     role     |  count  
--------------+---------
              |       1
 1            | 1076075
 2            |  734425
 3            |  782212
 4            |  934818
 Accountable  |     280
 Extending    |   14628
 Funding      |   27480
 Implementing |   18304
(9 rows)


```

**Comment by odscjames** [comment]
We'll start by normalising Role data only to test approach. Store V2 Integer codes in DB.

Probably have to do in multi stages - do data mapping, then change column type.

Have to change tests too.

**Comment by odscjames** [comment]
@markbrough will update documentation
